File: cpp/samchon/library/ProgressEvent.cpp
#include <samchon/library/ProgressEvent.hpp>
using namespace samchon::library;

ProgressEvent::ProgressEvent(EventDispatcher *source, size_t numerator, size_t denominator)
	: Event(source, type)
{
	this->numerator = numerator;
	this->denominator = denominator;
}

auto ProgressEvent::getNumerator() const -> size_t
{
	return numerator;
};
auto ProgressEvent::getDenominator() const -> size_t
{
	return denominator;
};
auto ProgressEvent::getPercent() const -> double
{
	return numerator / (double)denominator;
};

File: ts/samchon/library/ProgressEvent.ts
/// <reference path="BasicEvent.ts" />

namespace samchon.library
{
    export class ProgressEvent
        extends BasicEvent
    {
        public static get PROGRESS(): string { return "progress"; }

        protected numerator_: number;
        protected denominator_: number;

        public constructor(type: string, numerator: number, denominator: number)
        {
            super(type);

            this.numerator_ = numerator;
            this.denominator_ = denominator;
        }

        public get numerator(): number
        {
            return this.numerator_;
        }
        public get denominator(): number
        {
            return this.denominator_;
        }
    }
}